src/components/Contexts/StepperDialogProvider.jsx

```diff
@@ -69,23 +70,22 @@ const StepperDialogProvider = ({ children }) => {
       currentStepIndex,
       stepperDialogTitle,
       currentDefinition,
+      alreadyScan,
       setIsStepperDialogOpen,
       setCurrentStepIndex,
       setCurrentDefinition,
       setStepperDialogTitle,
       previousStep,
-      nextStep
+      nextStep,
+      setAlreadyScan
     }),
     [
+      isStepperDialogOpen,
       allCurrentSteps,
-      currentDefinition,
       currentStepIndex,
-      isStepperDialogOpen,
       stepperDialogTitle,
-      setIsStepperDialogOpen,
-      setCurrentStepIndex,
-      setCurrentDefinition,
-      setStepperDialogTitle,
```

Comment:
why are we removing all those methods?

Reply:
Just because it's useless & not recommended. See Note here: https://reactjs.org/docs/hooks-reference.html#usestate